### PR TITLE
Features/5277 infer delivery mode

### PR DESCRIPTION
**Pull request by utas-raymondng**

### CODE REVIEW

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        @Override
+        public String toString() {
+            return mode;
+        }
+    }
+
+    protected static Optional<DeliveryMode> scanRealTimeOrDelayed(String text) {
+        if (text != null) {
+            text = text.toLowerCase();
+            if(text.contains("real time") || text.contains("realtime")) {
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:29]: what about `real-time` ?

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+                return Optional.of(DeliveryMode.real_time);
+            }
+            else if(text.contains("delayed")) {
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:32]: also incorrect spelling e.g `delay` or `delayedd` ?

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    public static DeliveryMode getDeliveryMode(final MDMetadataType source) {
+        String status = SummariesUtils.getStatus(source);
+
+        if(status != null) {
+            status = status.toLowerCase();
+
+            if(status.contains("completed")) {
+                return DeliveryMode.completed;
+            }
+            else if(status.contains("ongoing")) {
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:67]: and `on-going`

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+     *                  Else - OTHER
+     * @return - The delivery mode enum
+     */
+    public static DeliveryMode getDeliveryMode(final MDMetadataType source) {
+        String status = SummariesUtils.getStatus(source);
+
+        if(status != null) {
+            status = status.toLowerCase();
+
+            if(status.contains("completed")) {
```
Comment (vietnguyengit) [inline indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:64]: or `complete` or `finished` etc.

**Comment by vietnguyengit** [comment]
I dont know the library in Java to fuzzy group similar texts, in Python, I have this script: https://github.com/aodn/geonetwork4-build/blob/main/utilities/geonetwork-gcmd-extractor/utils/nlp_grouping.py

**Comment by vietnguyengit** [comment]
the effect looks like: https://github.com/aodn/geonetwork4-build/tree/main/utilities/geonetwork-gcmd-extractor#nlp-grouping-similar-texts

**Comment by vietnguyengit** [comment]
but it's really up to you to consider, I didn't tick "requested changes" for this PR tho.

**Review by vietnguyengit — APPROVED** [review]
approved with comments

**Comment by utas-raymondng** [comment]
> the effect looks like: https://github.com/aodn/geonetwork4-build/tree/main/utilities/geonetwork-gcmd-extractor#nlp-grouping-similar-texts

how to use it with java code? May be a new ticket to deal with it, I wonder if this way of identify real time will change

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java (hunk 3 above)
Comment (utas-raymondng) [reply indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:67]: no it is ongoing in xml

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java (hunk 2 above)
Comment (utas-raymondng) [reply indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:32]: it is delayed

File: indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java (hunk 1 above)
Comment (utas-raymondng) [reply indexer/src/main/java/au/org/aodn/esindexer/utils/DeliveryModeUtils.java:29]: Updated

**Review by vietnguyengit — APPROVED** [review]

**Comment by vietnguyengit** [comment]
> > the effect looks like: https://github.com/aodn/geonetwork4-build/tree/main/utilities/geonetwork-gcmd-extractor#nlp-grouping-similar-texts
> 
> how to use it with java code? May be a new ticket to deal with it, I wonder if this way of identify real time will change

yeah, probably need to find alternative java dependencies or somehow to port that python code.

**Comment by utas-raymondng** [comment]
Followup action in this ticket https://github.com/aodn/backlog/issues/5847

**Pull request merged**